crates/matrix-sdk/src/room/timeline/tests.rs

```diff
@@ -542,6 +544,178 @@ async fn initial_events() {
     assert_eq!(items[2].as_event().unwrap().sender(), *BOB);
 }
 
+#[async_test]
+async fn other_state() {
+    let timeline = TestTimeline::new(&ALICE);
+    let mut stream = timeline.stream();
+
+    timeline
+        .handle_live_custom_event(json!({
```

Comment:
Instead of `_custom_event`, can you add `handle_live_state_event` (for `EmptyStateKey`) and `handle_live_state_event_with_state_key`, akin to `handle_live_message_event` and use that?